experimental/ir/lower_imports.go

```diff
@@ -109,3 +100,23 @@ func buildImports(f File, r *report.Report, importer Importer) {
 		}
 	}
 }
+
+func diagnoseCycle(r *report.Report, path string, cycle ErrCycle[ast.DeclImport]) {
+	err := r.Errorf("encountered cycle while importing %q", path)
+
+	for i, imp := range cycle {
+		var message string
+		path, ok := imp.ImportPath().AsLiteral().AsString()
+		if ok {
+			switch i {
+			case 0:
+				break
```

Comment:
I know you added a comment to the `ErrCycle` type, but it would be useful here, too, when scanning this method body -- that we skip the first item because it's the same/redundant with the last one.